### PR TITLE
Prevent panic when copying text outside of a secure context

### PR DESCRIPTION
**Pull request by YgorSouza**
* Closes <https://github.com/emilk/egui/issues/5293>
* [x] I have followed the instructions in the PR template

### CODE REVIEW

File: crates/eframe/src/web/mod.rs [hunk 1]
```diff
@@ -174,6 +174,13 @@ fn set_cursor_icon(cursor: egui::CursorIcon) -> Option<()> {
 /// Set the clipboard text.
 fn set_clipboard_text(s: &str) {
     if let Some(window) = web_sys::window() {
+        if !window.is_secure_context() {
+            log::error!(
```
Comment (rustbasic) [inline crates/eframe/src/web/mod.rs:178]:
```suggestion
            log::debug!(
```

Comment (lucasmerlin) [reply crates/eframe/src/web/mod.rs:178]: I think it's fine to log at least a warning here. Something is clearly wrong when this occurs and the developer should have a chance to see this error message so they know what's wrong.

**Comment by lucasmerlin** [comment]
Thank you!

Maybe we should also update the [copy_text](https://docs.rs/egui/latest/egui/struct.Context.html#method.copy_text) docs to mention that this might not work for wasm builds outside of secure contexts / when the app is deployed without https?

**Review by emilk — APPROVED** [review]

**Comment by YgorSouza** [comment]
> Thank you!
> 
> Maybe we should also update the [copy_text](https://docs.rs/egui/latest/egui/struct.Context.html#method.copy_text) docs to mention that this might not work for wasm builds outside of secure contexts / when the app is deployed without https?

I added it, although I don't know if this is the right place for that, since it doesn't really have anything to do with egui, it's just how wasm works. But I guess at least it is hard to miss.

**Pull request merged**